### PR TITLE
Implement JSON serialization protocol for common channels

### PR DESCRIPTION
**Pull request by kevinsung**

### CODE REVIEW

File: cirq/ops/common_channels.py [hunk 1]
```diff
@@ -89,6 +89,21 @@ def _circuit_diagram_info_(self,
...
+    @property
+    def p_x(self):
+        return self._p_x
+
+    @property
+    def p_y(self):
+        return self._p_y
+
+    @property
+    def p_z(self):
```
Comment (Strilanc) [inline cirq/ops/common_channels.py]: Add a docstring and type annotation to these methods.

File: cirq/ops/common_channels.py [hunk 2]
```diff
@@ -164,6 +179,13 @@ def _circuit_diagram_info_(self,
                                args: 'protocols.CircuitDiagramInfoArgs') -> str:
         return 'D({!r})'.format(self._p)
 
+    @property
+    def p(self):
```
Comment (Strilanc) [inline cirq/ops/common_channels.py]: Add a docstring and type annotation to this method.

File: cirq/ops/common_channels.py [hunk 3]
```diff
@@ -289,6 +311,17 @@ def _circuit_diagram_info_(self,
                                args: 'protocols.CircuitDiagramInfoArgs') -> str:
         return 'GAD({!r},{!r})'.format(self._p, self._gamma)
 
+    @property
+    def p(self):
```
Comment (Strilanc) [inline cirq/ops/common_channels.py]: Add a docstring and type annotation to these methods.

File: cirq/ops/common_channels.py [hunk 4]
```diff
@@ -406,6 +439,13 @@ def _circuit_diagram_info_(self,
                                args: 'protocols.CircuitDiagramInfoArgs') -> str:
         return 'AD({!r})'.format(self._gamma)
 
+    @property
+    def gamma(self):
```
Comment (Strilanc) [inline cirq/ops/common_channels.py]: Add a docstring and type annotation to this method.

File: cirq/ops/common_channels.py [hunk 5]
```diff
@@ -513,6 +553,13 @@ def _circuit_diagram_info_(self,
                                args: 'protocols.CircuitDiagramInfoArgs') -> str:
         return 'R'
 
+    @property
+    def dimension(self):
```
Comment (Strilanc) [inline cirq/ops/common_channels.py]: Add a docstring and type annotation to this method.

File: cirq/ops/common_channels.py [hunk 6]
```diff
@@ -585,6 +632,13 @@ def _circuit_diagram_info_(self,
                                args: 'protocols.CircuitDiagramInfoArgs') -> str:
         return 'PD({!r})'.format(self._gamma)
 
+    @property
+    def gamma(self):
```
Comment (Strilanc) [inline cirq/ops/common_channels.py]: Add a docstring and type annotation to this method.

File: cirq/ops/common_channels.py [hunk 7]
```diff
@@ -682,6 +736,13 @@ def _circuit_diagram_info_(self,
                                args: 'protocols.CircuitDiagramInfoArgs') -> str:
         return 'PF({!r})'.format(self._p)
 
+    @property
+    def p(self):
```
Comment (Strilanc) [inline cirq/ops/common_channels.py]: Add a docstring and type annotation to this method.

File: cirq/ops/common_channels.py [hunk 8]
```diff
@@ -827,6 +888,13 @@ def _circuit_diagram_info_(self,
                                args: 'protocols.CircuitDiagramInfoArgs') -> str:
         return 'BF({!r})'.format(self._p)
 
+    @property
+    def p(self):
```
Comment (Strilanc) [inline cirq/ops/common_channels.py]: Add a docstring and type annotation to this method.

**Comment by kevinsung** [comment]
Done.

**Comment by mpharrigan** [comment]
woo

**Comment by kevinsung** [comment]
Windows build failed with no output.

**Comment by Strilanc** [comment]
That's the threading flake that @c-poole is working on. Just re-start it.

**Pull request merged**